lib/hubspot/config.rb

```diff
@@ -14,7 +18,18 @@ def configure(config)
         @hapikey = config["hapikey"]
         @base_url = config["base_url"] || "https://api.hubapi.com"
         @portal_id = config["portal_id"]
-        @logger = config['logger'] || DEFAULT_LOGGER
+        @logger = config["logger"] || DEFAULT_LOGGER
+        @access_token = config["access_token"]
+        @client_id = config["client_id"] if config["client_id"].present?
+        @client_secret = config["client_secret"] if config["client_secret"].present?
+        @redirect_uri = config["redirect_uri"] if config["redirect_uri"].present?
+
+        unless access_token.present? ^ hapikey.present?
```

Comment:
I don't think you want to #present? as it's a rails specific method? Also, I think the logical OR (||) is more appropriate. This should do it:

`unless access_token || hapikey`